windows-containers/understanding-windows-containers.adoc

```diff
@@ -0,0 +1,51 @@
+[id="understanding-windows-containers"]
+= Understanding Windows containers
+include::modules/common-attributes.adoc[]
+:context: understanding-windows-containers
+
+toc::[]
+
+{product-title} provides built-in support for running Microsoft Windows Server
+containers. For those that administer heterogeneous environments with a mix of
+Linux and Windows workloads, {product-title} allows you to deploy Windows
+workloads running on Windows Server containers while also providing traditional
+Linux workloads hosted on {op-system-first} or {op-system-base-full}.
+
+To run Windows workloads in your cluster, you must first install the Windows
+Machine Config Operator (WMCO). The WMCO orchestrates the process of deploying
+and managing Windows workloads on a cluster.
+
+Before deploying Windows workloads, you must create a Windows compute node and
+have it join the cluster. The Windows node hosts the Windows workloads in a
+cluster, and can run alongside other Linux-based compute nodes. You
+can create a Windows compute node by creating a Windows MachineSet to host
+Windows Server compute machines. You must apply a Windows-specific label to
+these machines that specifies a Windows OS image that has the Docker container
+runtime add-on enabled.
+
+The WMCO watches for machines with the Windows label. Once a Windows MachineSet
+is detected and provisioned, the WMCO configures the underlying Windows VM so
+that it can join the cluster as a compute node.
+
+The WMCO expects a predetermined secret in its namespace containing a private
+key that is used to interact with the Windows instance. WMCO checks for this
+secret during boot up time and creates a user data secret which you must
+reference in the Windows MachineSet that you created. Then the WMCO populates
+the user data secret with a public key that corresponds to the private key. With
+this data in place, the cluster can connect to the Windows VM using an SSH
+connection.
+
+From there, you can schedule Windows workloads to the Windows node.
```

Comment:
Just adding this for when you start filling this out in the future... this is where I would suggest talking about how the scheduling happens: [Guide for scheduling Windows containers in Kubernetes](https://kubernetes.io/docs/setup/production-environment/windows/user-guide-windows-containers/). Namely call out [Simplifying with RuntimeClass](https://kubernetes.io/docs/setup/production-environment/windows/user-guide-windows-containers/#simplifying-with-runtimeclass)

Reply:
@aravindhp looking into this now. How do you propose we describe `RuntimeClass` in this context? Is this something that we need a procedure covering? Or is this more for informational background for how WMCO schedules workloads behind-the-scenes?

Reply:
We should mention `RuntimeClass` in context of simplifying how to use `Taints/Toleration` and  `NodeSelectors`. It also allows them to easily handle multiple Windows OS variants (there would be a Windows MachineSet for each OS variant) as Windows containers needs to be mapped to a base Windows OS variant. Please see [Windows host and container version compatibility](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/update-containers#host-and-container-version-compatibility).

Reply:
@aravindhp after looking into this a bit today... does recommending a `RuntimeClass` interfere with the support statement that we only support Windows workloads on Windows Server 2019? I get the impression a `RuntimeClass` would be useful if we supported more Windows OS variants, but with us only supporting one, is this applicable to the current state of Windows containers in OCP? Or am I mixing terms here?

Reply:
Or is this to differentiate between Linux and Windows? :thinking: